source/common/version/api_version.h

```diff
@@ -0,0 +1,30 @@
+#pragma once
+
+#include <string>
+
+#include "common/version/api_version_struct.h"
+
+// Defines the ApiVersion version (Envoy::api_version).
+#include "common/version/api_version_number.h"
+
+namespace Envoy {
+
+class ApiVersionInfoTestPeer;
+
+/**
+ * Wraps compiled in api versioning.
+ */
+class ApiVersionInfo {
+public:
+  // Returns the most recent API version that is supported by the client.
+  static const ApiVersion& apiVersion();
+
+  // Returns the oldest API version that is supported by the client.
+  static const ApiVersion& oldestApiVersion();
```

Comment:
I think these two can be `constexpr` and no need of `computeOldestApiVersion()`.

Reply:
I moved the function to the python script.
RE `constexpr` I think having the functions (with comments) here is better, otherwise the variables will have no comments in an IDE. WDYT?

Reply:
I mean't they can be `constexpr` functions

Reply:
Done.